### ISSUE TITLE
[packchk] Add validation that a source module cannot be concurrently added to a project by multiple components

### ISSUE THREAD

**Issue by jkrech** [Open-CMSIS-Pack/devtools#1499]
**Describe The Problem To Be Solved**
For a successful build of a project all source modules must be unique. In case a project selects two components referencing the identical source module this can lead to duplicated objects causing errors at link time and is considered incorrect software design.

**Suggest A Solution**
Add a validation step for all component's `<file>` elements with attribute `category=` in (library, object, source, sourceC, sourceCpp, sourceAsm, genSource) 
a) finding identical references to files in the pack
b) checking whether they belong to component's that can be selected concurrently
- file conditions are mutually exclusive
- components' conditions are mutually exclusive
- components belong to different `Cbundle`
- components belong to different `Cvariant`

**Comment by jkrech** [comment]
Note: this is detected by `csolution` if components are selected concurrently.

**Comment by edriouk** [comment]
I would suggest co check if components having the same files belong to the same Cclass/Cgroup/Csub, but different bundle or variant. Otherwise issue a warning that different components refer to the same files.

**Comment by thorstendb-ARM** [comment]
Currently discussing with Evgueni if it is possible to let RTE Model filter with respect to all bundles, components, variants and conditions, to let it return a list that can be checked.

**Comment by jkrech** [comment]
I much rather accept some sloppyness in the packchk validation than relying on developers finding it only when concurrently selecting compontents and seeing the warning in the `csolution` output.
If we find two componentIDs within a pack matching but having two different `condition=` strings, these components cannot be concurrently selected.
This means regarding conditions we can really focus on file conditions.
If we find the reference to the same file in two components that are concurrently selectable (components differing only in bundle or variant or condition, cannot be concurrently selected) and the common file references have a different `condition=` string we simply assume (hence sloppy) the conditions are such that they will not be in the project at the same time and do not issue a warning.

**Comment by edriouk** [comment]
I would issue a warning anyway if components referring to the same file belong to different taxonomies (Cclass/Cgroup).

**Comment by ReinhardKeil** [comment]
@edriouk I agree with you. I think as long as we use Cclass/Cgroup this should work fine.

**Comment by thorstendb-ARM** [comment]
> but having two different condition= strings
Conditions must be evaluated, packchk cannot rely on "different strings".

**Comment by brondani** [comment]
Validating `packchk 1.4.2`:
Complementary to the comment:
>  the common file references have a different condition= string we simply assume (hence sloppy) the conditions are such that they will not be in the project at the same time and do not issue a warning

it can be inferred a file without a `condition=` will be in the project and a warning for common references should be issued.

For example in the following test case the files [`Files/system_Test.c`](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/packchk/test/data/ConcurrentComponentFiles/ARM.ConcurrentComponentFiles.pdsc#L70) (lines 70 and 78) can be potentially concurrent but it is not reported by packchk:
```
<file category="source"  name="Files/system_Test.c"      attr="config" version="1.0.0" />
```
